src/App.tsx

```diff
@@ -14,37 +15,107 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+  None = '',
+}
 
 export const App: React.FC = () => {
+  const [sortedGoods, setSortedGoods] = useState<string[]>([
+    ...goodsFromServer,
+  ]);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const sortGoods = (goods: string[], type: SortType): string[] => {
+    const sorted = [...goods];
+
+    switch (type) {
+      case SortType.Alphabetical:
+        sorted.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.Length:
+        sorted.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+
+    if (isReversed) {
+      sorted.reverse();
+    }
+
+    return sorted;
+  };
+
+  const handleSort = (type: SortType) => {
+    const newSortType = type || sortType;
+
+    setSortType(newSortType);
+
+    const sorted = sortGoods(goodsFromServer, newSortType);
+
+    setSortedGoods(sorted);
+  };
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
+    setSortedGoods(prevGoods => [...prevGoods].reverse());
+  };
+
+  const resetSort = () => {
+    setSortType(SortType.None);
+    setIsReversed(false);
+    setSortedGoods([...goodsFromServer]);
+  };
+
+  const buttonStyle = (type: SortType) => (sortType === type ? '' : 'is-light');
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', buttonStyle(SortType.Alphabetical))}
+          onClick={() => handleSort(SortType.Alphabetical)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', buttonStyle(SortType.Length))}
+          onClick={() => handleSort(SortType.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', { 'is-light': !isReversed })}
```

Comment:
The active state for the 'Reverse' button is not clearly marked for the user. According to checklist item 3, you should indicate the active state. Consider using a different style or class when the list is reversed to make it clear to the user that the action is active.